services/src/main/java/org/adamalang/services/social/Discord.java

```diff
@@ -22,46 +22,227 @@
 import org.adamalang.common.Callback;
 import org.adamalang.common.ErrorCodeException;
 import org.adamalang.common.Json;
+import org.adamalang.common.XWWWFormUrl;
+import org.adamalang.common.metrics.RequestResponseMonitor;
 import org.adamalang.runtime.natives.NtPrincipal;
 import org.adamalang.runtime.remote.ServiceConfig;
 import org.adamalang.runtime.remote.SimpleService;
 import org.adamalang.metrics.FirstPartyMetrics;
-import org.adamalang.web.client.WebClientBase;
+import org.adamalang.web.client.*;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import java.io.ByteArrayOutputStream;
+import java.nio.charset.StandardCharsets;
 import java.util.HashSet;
+import java.util.Locale;
+import java.util.TreeMap;
 import java.util.function.Consumer;
 
 public class Discord extends SimpleService {
   private static final Logger LOGGER = LoggerFactory.getLogger(Discord.class);
   private final FirstPartyMetrics metrics;
   private final WebClientBase base;
+  private final String botToken;
 
-  public Discord(FirstPartyMetrics metrics, WebClientBase base) {
+  private final String userToken;
+
+
+  public Discord(FirstPartyMetrics metrics, WebClientBase base, String botToken, String userToken) {
     super("discord", new NtPrincipal("discord", "service"), true);
     this.metrics = metrics;
     this.base = base;
+    this.botToken = botToken;
+    this.userToken = userToken;
   }
 
   public static Discord build(FirstPartyMetrics metrics, ServiceConfig config, WebClientBase base) throws ErrorCodeException {
-    return new Discord(metrics, base);
+    return new Discord(metrics, base, config.getString("bot_token", ""), config.getString("user_token", ""));
   }
 
+  class DiscordHttpResponder implements SimpleHttpResponder {
```

Comment:
Why implement this rather than using StringHttpResponder ?

Reply:
I thought that maybe some logic in StringHttpResponder would not be good for the initial test, primarily the if status code is successful, but it ended up looking basically the same as the original. 

In the latest push, I  just used StringCallbackHttpResponder